### PR TITLE
Lkt: emit an error if PropertyError is called without raise

### PR DESCRIPTION
**Pull request by thvnx**
Ensure that PropertyError is called through the raise keyword.

TN: U928-029

### CODE REVIEW

File: contrib/lkt/language/parser.py [hunk 1]
```diff
@@ -447,6 +447,17 @@ def check_semantic_impl():
             has_error=results.any(lambda r: r.has_error)
         )
 
+    @langkit_property(return_type=T.LKNode.entity)
+    def first_no_paren_parent():
+        """
+        Return the first parent that is not a ParenExpr.
```
Comment (pmderodat) [inline contrib/lkt/language/parser.py:453]: Please add double backquotes around code (`ParenExpr`).

File: contrib/lkt/language/parser.py [hunk 2]
```diff
@@ -2449,6 +2460,23 @@ def check_correctness_pre():
...
+                If(
+                    And(
+                        rd.result_ref == Entity.property_error_type,
+                        Not(Entity.first_no_paren_parent.is_a(T.RaiseExpr))
+                    ),
+
+                    Self.error(
+                        S("cannot call ").concat(
+                            Entity.property_error_type.full_name
+                        ).concat(S(" without a raise expression"))
```
Comment (pmderodat) [inline contrib/lkt/language/parser.py:2474]: To be more specific, I would reword: “cannot call … outside of a raise expression”.

**Review by pmderodat — CHANGES_REQUESTED** [review]

**Review by pmderodat — CHANGES_REQUESTED** [review]

File: testsuite/tests/contrib/lkt_semantic/property_error/test.lkt [hunk 3]
```diff
@@ -1,5 +1,7 @@
 fun prop1(): Bool = raise PropertyError("error")
 fun prop2(): Bool = (raise PropertyError("error"))
+fun prop21(): Bool = raise (PropertyError("error"))
+fun prop22(): Bool = raise (((PropertyError("error"))))
```
[the remark this answers is not in the document]
Comment (thvnx) [reply testsuite/tests/contrib/lkt_semantic/property_error/test.lkt:4]: OK, it needs a bit more of work then! I'm on it.

[the remark this answers is not in the document]
Comment (raph-amiard) [reply testsuite/tests/contrib/lkt_semantic/property_error/test.lkt:4]: @pmderodat @thvnx I think Laurent can do that in a separate PR. In general let's try to keep PRs as small as possible since they have a tendency to linger (ie. not add new cases to existing functional PRs)

[the remark this answers is not in the document]
Comment (thvnx) [reply testsuite/tests/contrib/lkt_semantic/property_error/test.lkt]: I've just added a TODO for that.

**Review by pmderodat — DISMISSED** [review]

**Review by pmderodat — APPROVED** [review]
